packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -291,8 +294,8 @@ export class RestApi extends Resource implements IRestApi {
    * @param id The construct id
    * @param options custom domain options
    */
-  public addDomainName(id: string, options: DomainNameOptions): DomainName {
-    return new DomainName(this, id, {
+  public addDomainName(id: string, options: DomainNameOptions): void {
+    this._domainName = new DomainName(this, id, {
       ...options,
       mapping: this
     });
```

Comment:
This change will break existing customers when used in a specific situation.

Consider this CDK app.

```ts
  import { ApiGateway } from '@aws-cdk/aws-route53-targets';
  import { RestApi } from '@aws-cdk/aws-apigateway';

  const restApi = new RestApi(stack, 'id', {
    domainName: {
      domainName: 'first-domain'
      ...
  });

  restApi.addDomainName('second-domain', {
    domainName: 'second-domain',
    ...
  });

  const domainName = restApi.domainName;
```

Before this change, `domainName` would point to `first-domain`; while, after this change, `domainName` will now point to `second-domain`.

This needs to be fixed.

Further, add another unit test to assert this and update the documentation of `get domainName()` to state this.

Reply:
didn't even think about that. 
Thanks for the review, I'll send an update later.